tests/test_edgeql_calls.py

```diff
@@ -1141,3 +1141,137 @@ class TestEdgeQLFuncCalls(tb.QueryTestCase):
                 [1, 2],
             ]
         ])
+
+    @test.xfail('''
+        The operator IS is normally static, so special care needs to
+        be taken when it's used inside polymorphic functions.
+    ''')
+    async def test_edgeql_calls_33(self):
+        await self.con.execute('''
+            CREATE FUNCTION test::call33(
+                a: anytype
+            ) -> bool
+                FROM EdgeQL $$
+                    SELECT a IS float32
```

Comment:
We should just prohibit using `IS` in polymorphic functions on polymorphic types.  We have no ETA when (if ever) we can add support for this.  So instead of xfail, please make it raise a proper exception.

Reply:
I'll make a separate small PR forbidding `IS` with polymorphic types.